lib/package.gi

```diff
@@ -1958,6 +1978,35 @@ InstallGlobalFunction( ExtendRootDirectories, function( rootpaths )
     end );
 
 
+#############################################################################
+##
+#F  ExtendPackageDirectories( <paths_or_dirs> )
+##
+InstallGlobalFunction( ExtendPackageDirectories, function( paths_or_dirs )
```

Comment:
Have you tested this function, at least superficially? 

Actually, perhaps we can test it by using it in our test suite: if you look at `tst/testinstall/package.tst` it currently has this hack:
```
# load mockpkg via a symlink in a directory called `pkg`
# so we can test ExtendRootDirectories below
# first create a temporary directory for all of this
gap> tmp_dir := DirectoryTemporary( );;

# create a subdirectory `<tmp_dir>/pkg`
gap> Exec( Concatenation( "mkdir -p ", Filename( tmp_dir, "/pkg" ) ) );

# make `<tmp_dir>/pkg/mockpkg` a symlink to `tst/mockpkg`
gap> Exec( Concatenation( "ln -sfn ", Filename( DirectoriesLibrary("tst/mockpkg"), "" )," ", Filename( tmp_dir, "pkg/mockpkg" ) ) );
```

If we could use `ExtendPackageDirectories` here it would serve both as a test for `ExtendPackageDirectories` and also get rid of this ugly hack...

Reply:
I am not quite sure what you are suggesting here. Yes, we could remove this hack here and replace the call to `ExtendRootDirectories` in https://github.com/gap-system/gap/blob/1c45e6fe4e8c23630c6b4c9be6b996587595df0d/tst/testinstall/package.tst#L641 by `ExtendPackageDirectories`. However, this would get rid of the only test of `ExtendPackageDirectories`. Is this what you would like to do here?

Reply:
We can easily write a different test for `ExtendRootDirectories` by just pointing it at *any* directory with a "unique" file(name) in it and then trying to query that. But this is the one place where `ExtendPackageDirectories` is perfect